bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/widgets/ScaleRenderer.java

```diff
@@ -0,0 +1,225 @@
+			lastNotch = bar.y + bar.height - 5;
+		} else {
+			bar = new Rectangle(x + 8, y + 19, w - (x + 8) - 7 - fix, 4 - fix);
+			firstNotch = bar.x + 5;
+			lastNotch = bar.x + bar.width - 5;
+		}
+
+		gc.fillRectangle(bar);
+		gc.setForeground(scale.getDisplay().getSystemColor(SWT.COLOR_WIDGET_NORMAL_SHADOW));
+		gc.drawRectangle(bar);
+
+		// prepare for line drawing
+		gc.setForeground(scale.getDisplay().getSystemColor(SWT.COLOR_WIDGET_NORMAL_SHADOW));
+		gc.setLineWidth(1);
+		gc.setLineWidth(1);
+
+		// draw first and last notch
+		drawNotch(gc, firstNotch, 4);
+		drawNotch(gc, lastNotch, 4);
+
+		// draw center notches
+		int unitPerPage = scale.getPageIncrement();
+		double totalPixel = lastNotch - firstNotch;
+		ppu = totalPixel / units;
+		drawCenterNotches(gc, firstNotch, lastNotch, units, unitPerPage, totalPixel);
+
+		drawHande(gc, effectiveValue);
+	}
+
+	private void drawCenterNotches(IGraphicsContext gc, int firstNotchPos, int lastNotchPos, int units, int unitPerPage,
+			double scalePixels) {
+		if (isRTL() && !isVertical()) {
+			for (int i = unitPerPage; i < units; i += unitPerPage) {
+				int position = lastNotchPos - (int) (i * ppu);
+				drawNotch(gc, position, 3);
+			}
+		} else { // SWT.LEFT_TO_RIGHT or SWT.VERTICAL
+			for (int i = unitPerPage; i < units; i += unitPerPage) {
+				int position = firstNotchPos + (int) (i * ppu);
+				drawNotch(gc, position, 3);
+			}
+		}
+	}
+
+	private void drawHande(IGraphicsContext gc, int value) {
+		// draw handle
+		Color handleColor = switch (scale.getHandleState()) {
+		case IDLE -> IDLE_COLOR;
+		case HOVER -> HOVER_COLOR;
+		case DRAG -> DRAG_COLOR;
+		};
+		gc.setBackground(handleColor);
+		handleBounds = calculateHandleBounds(value);
+		gc.fillRectangle(handleBounds);
+	}
+
+	private Rectangle calculateHandleBounds(int value) {
+		int pixelValue = (int) (ppu * value);
+		if (isVertical()) {
+			return new Rectangle(bar.x - 9, bar.y + pixelValue, 21, 10);
+		} else if (isRTL()) {
+			return new Rectangle(bar.x + bar.width - pixelValue - 10, bar.y - 9, 10, 21);
+		} else {
+			return new Rectangle(bar.x + pixelValue, bar.y - 9, 10, 21);
+		}
+	}
+
+	@Override
+	public int handlePosToValue(Point pos) {
+		if (isVertical()) {
+			return (int) Math.round((pos.y - bar.y) / ppu);
+		} else {
+			return (int) Math.round((pos.x - bar.x) / ppu);
+		}
+	}
+
+	private boolean isRTL() {
+		return scale.getOrientation() == SWT.RIGHT_TO_LEFT;
+	}
+
+	private boolean isVertical() {
+		return scale.getAlignement() == SWT.VERTICAL;
+	}
+
+	private void drawNotch(IGraphicsContext gc, int pos, int size) {
+		if (isVertical()) {
+			gc.drawLine(bar.x - 10 - size, pos, bar.x - 10, pos);
+			gc.drawLine(bar.x + 14, pos, bar.x + 14 + size, pos);
+		} else {
+			gc.drawLine(pos, bar.y - 10 - size, pos, bar.y - 10);
+			gc.drawLine(pos, bar.y + 14, pos, bar.y + 14 + size);
+		}
+	}
+
+	// TODO move this into the GC itself
+	private int getGCCorrectionValue() {
+		return SWT.USE_SKIJA ? 0 : 1;
+	}

```

Comment:
Just saw this TODO and still having this GC-dependent correction (like in `Button`) is definitely not good. I have provided a PR to make the behavior of SkijaGC conform with the one of the original GC:
- #32

After that PR, this correction calculation can be removed completely (and, if necessary, replaced by a unified constant).

The `Scale` widget also shows a miscalculation regarding filled regions, as the handle appear blurry:
![image](https://github.com/user-attachments/assets/ccccc396-f655-4c32-939c-c298fbff2648)

With the mentioned PR this should be fixed as well.